code/datums/components/chameleon/spritesheet.dm

```diff
@@ -0,0 +1,44 @@
+/datum/asset/spritesheet/chameleon
+	name = "chameleon"
+
+/datum/asset/spritesheet/chameleon/create_spritesheets()
+	var/list/disguises = list()
+	// First, get everything we need to make icons of.
+	for(var/chameleon_path in subtypesof(/datum/component/chameleon))
+		var/obj/item/dummy = new
+		var/datum/component/chameleon/chameleon = dummy.AddComponent(chameleon_path)
+		if(!chameleon.base_disguise_path && !chameleon.disguise_whitelist)
+			qdel(dummy)
+			continue
+		disguises |= list_chameleon_disguises(chameleon.base_disguise_path, typecacheof(chameleon.disguise_whitelist), typecacheof(chameleon.disguise_blacklist, only_root_path = TRUE), chameleon.hide_duplicates)
+		qdel(dummy)
+	// Then, we need to generate the actual icons.
+	for(var/item_path in disguises)
+		add_item(item_path)
+	// I like this better than hardcoding a list of all chameleon items.
+	for(var/item_path in subtypesof(/obj/item))
+		var/item_path_txt = "[item_path]"
+		if(!findtext(item_path_txt, "/chameleon") || findtext(item_path_txt, "/broken") || (item_path in disguises))
```

Comment:
use ispath() at least

Reply:
This was just because there were multiple _separate_ subtypes that ended up with `/chameleon` and `/broken`, and I didn't want to hardcode those types... and I don't think `ispath` can check only the last part of a path?